File: 134-gas-station/gas-station.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {

        int n=gas.length;

        int sum_gas=0;
        int sum_cost=0;
        for(int i=0 ;i< n; i++){
            sum_gas+=gas[i];
            sum_cost+=cost[i];
        }
        if(sum_cost-sum_gas> 0) return -1;

        int total=0;
        int res=0;
        for(int i=0; i< n ;i++){

            total+= (gas[i]-cost[i]);
            if(total<0){
                total=0;
                res=i+1;
            }

        }
        return res;
        
    }
}

File: 134-gas-station/gas-station.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        total_surplus,surplus=0,0
        start=0
        n=len(gas)
        for i in range(0,n):
            gain=gas[i]-cost[i]
            total_surplus+=gain
            surplus+=gain

            if(surplus<0):
                surplus=0
                start=i+1
        
        return -1 if total_surplus<0 else start

